price-feeder/oracle/provider/binance.go

```diff
@@ -145,13 +161,21 @@ func (p *BinanceProvider) getTickerPrice(key string) (TickerPrice, error) {
 	return ticker.toTickerPrice()
 }
 
-func (p *BinanceProvider) getTickerTrades(key string) (BinanceCandle, error) {
-	candle, ok := p.candles[key]
+func (p *BinanceProvider) getCandlePrices(key string) ([]CandlePrice, error) {
+	candles, ok := p.candles[key]
```

Comment:
`p.candles` isn't mutex protected here. Please go through and make sure any field that is write protected is also read protected :)

Reply:
added lock & unlocks for candles and tickers in 21f4e22466ddf16ee487c47d39e588303d7b579d 👍